apps/dashr-clinical-analytics/app.R

```diff
@@ -0,0 +1,492 @@
+library(dashR)
+library(dashCoreComponents)
+library(dashHtmlComponents)
+library(plotly)
+library(chron)
+library(dplyr)
+
+appName <- Sys.getenv("DASH_APP_NAME")
+if (appName != ""){
+  pathPrefix <- sprintf("/%s/", appName)
+
+  Sys.setenv(DASH_ROUTES_PATHNAME_PREFIX = pathPrefix,
+             DASH_REQUESTS_PATHNAME_PREFIX = pathPrefix)
+
+  setwd(sprintf("./app/apps/%s", appName))
+}
+
+app <- Dash$new()
+
+setwd(sprintf("./app/apps/%s", appName))
+
+df <- read.csv('./data/clinical_analytics.csv', header = TRUE, sep = ",")
+
+
+#list of clinics
+clinics = unique(df$Clinic.Name)
+clinic_list <- list()
+for (i in 1:length(clinics)){
+  clinic_list[[i]] <- list(label = clinics[i], value = clinics[i])
+}
+
+#list of check in times
+times = unique(df$Check.In.Time)
+times_list <- list()
+for (i in 1:length(clinics)){
+  times_list[[i]] <- list(label = times[i], value = times[i])
+}
+
+#list of all departments 
+deps = unique(df$Department)
+all_departments = list()
+for(i in 1:length(deps)){
+  all_departments[[i]] = toString(deps[[i]])
+}
+
+df$Admit.Source[df$Admit.Source==""] <- "Not Identified"
+
+df$Days.of.Wk = df$Check.In.Hour = df$Check.In.Time
```

Comment:
No typo, I created two extra columns called `Days of Wk` and `Check In Hour` which needed the info from the original column `Check In Time`. So I loaded each column with `Check In Time` info, then changed the other columns accordingly. If there is a smoother way of doing it let me know and I'll change it :)

Reply:
Ah, OK. In that case I suggest

- break up your assignment statements on multiple lines
- use the assignment operator `<-` instead of `=` (more readable, since `==` is a comparison operator)
- use lower case for variable names
- separate words in variables with `_` (underscores) instead of `.` (dots)

The reason for the last item above is that R performs something called S3/S4 method dispatch for generic functions. For example, the `plot` method has a special function for `density` and `lm` objects. Assuming `x` is a object of either class, `plot(x)` could call `plot.density(x)` or `plot.lm(x)` automatically.

To avoid confusion, it's good practice to reserve the `x.y` syntax for S3/S4 methods only.

Reply:
See here also: http://adv-r.had.co.nz/Style.html 🙂